### PR TITLE
Create a reusable detail table for reports and goals

### LINKED ISSUE #24332: [UI] Create a detail Angular component to use in Reports and Goals
### Problem Statement
As a user, I want to be able to send a template reference or an array of data to a detailed Angular component so that it can similarly display the data as the Figma file.

### Steps to Reproduce
N/A

### Acceptance Criteria

- The detail Angular component should be able to receive a template reference or an array of data as input.
- The data should be displayed in a similar way as the Figma file.
- The component should be thoroughly tested to ensure that it functions correctly.

### dotCMS Version
N/A

### Proposed Objective

The objective is to create a reusable Angular component that can be used to display detailed information in a visually appealing and consistent manner.

### Proposed Priority
Medium

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

- Figma design file [LINK](https://www.figma.com/file/fAoCk4swhiwplkU5GP21YD/Portlet-%2F-Experiments?node-id=1464-12048&t=TqrVA2dnHLCpJgCE-4)

### Assumptions & Initiation Needs

- Angular framework is being used
- Developers have access to the Figma design file

### Quality Assurance Notes & Workarounds
N/A

### Sub-Tasks & Estimates
N/A

### PR DESCRIPTION
**Pull request by oidacra**
### Proposed Changes

- New reusable component to show details

### Checklist
- [X] Tests
- [X] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
Removed the old component and replaced it with the new one.

### Screenshots

Report detail
Screenshot 2023-03-30 at 6 33 14 PM

Details in goals conditions
Screenshot 2023-03-30 at 6 33 25 PM

### CODE REVIEW

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/shared/directives/dot-string-template-outlet.directive.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    private shouldRecreateView = (dotStringTemplateOutlet: SimpleChange): boolean => {
+        let shouldOutletRecreate = false;
+        if (dotStringTemplateOutlet.firstChange) {
+            shouldOutletRecreate = true;
+        }
+
+        return shouldOutletRecreate;
+    };
```
Comment (hmoreras) [inline core-web/apps/dotcms-ui/src/app/portlets/shared/directives/dot-string-template-outlet.directive.ts:77]:
```suggestion
        return !!dotStringTemplateOutlet.firstChange;
    };
```

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/dot-experiments-reports.component.ts [hunk 2]
```diff
@@ -41,6 +46,45 @@ import { DotExperimentsUiHeaderComponent } from '@portlets/dot-experiments/share
 })
 export class DotExperimentsReportsComponent implements OnInit {
     vm$: Observable<VmReportExperiment> = this.store.vm$;
+    defaultVariantId = DEFAULT_VARIANT_ID;
+    detailData = [
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/dot-experiments-reports.component.ts:50]: Yes. Until the backend is ready.

File: core-web/apps/dotcms-ui/src/app/portlets/shared/directives/dot-string-template-outlet.directive.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+ * And `title` shows the `TemplateRef` or the `string` sent by the input
+ */
+
+@Directive({
+    standalone: true,
+    selector: '[dotStringTemplateOutlet]'
+})
+export class DotStringTemplateOutletDirective implements OnChanges {
+    @Input() dotStringTemplateOutlet: unknown | TemplateRef<unknown> = null;
+    private embeddedViewRef: EmbeddedViewRef<unknown> | null = null;
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/shared/directives/dot-string-template-outlet.directive.ts:38]: Yes, line 64, its contains the creation of the embebedView:

`this.embeddedViewRef = this.viewContainer.createEmbeddedView(
            templateRef as TemplateRef<unknown>,
            isTemplateRef ? this.dotStringTemplateOutlet : this.context
        );`

File: core-web/apps/dotcms-ui/src/app/portlets/shared/directives/dot-string-template-outlet.directive.ts (hunk 1 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/shared/directives/dot-string-template-outlet.directive.ts:77]: Nice, yes. Other cases exist, but right now, your suggestion works perfectly.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goals/dot-experiments-configuration-goals.component.html [hunk 4]
```diff
@@ -23,18 +23,43 @@ <h3 class="flex description" data-testId="goals-empty-msg">
...
-                            pButton
-                            pRipple
-                            type="button"
-                        ></button>
-                    </dot-experiments-experiment-goal-configuration-detail>
+                    <dot-experiments-details-table
+                        [data]="vm.goals.primary.conditions"
+                        [title]="headerTpl"
+                    >
+                        <ng-template #headers>
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goals/dot-experiments-configuration-goals.component.html:30]: Wrong name selection for the template; I will change it to `titleTpl`; the headers template  are the headers of the table (actually, I didn't use Table)

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/ui/dot-experiments-details-table/dot-experiments-details-table.component.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+    //** List of data to display, without templates, use the index of the objet as a header */
+    @Input()
+    data!: Array<unknown>;
+
+    //** Template to display the headers */
+    @ContentChild('headers', { static: true }) headers!: TemplateRef<unknown>;
+
+    //** Template to display the rows */
+    @ContentChild('rows', { static: true }) rows!: TemplateRef<unknown>;
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/ui/dot-experiments-details-table/dot-experiments-details-table.component.ts:39]: `@ContentChild` get a template variable from the content projection. In this case `rows` and `headers`, if not exist, I use the default templates.

File: core-web/apps/dotcms-ui/src/app/portlets/shared/directives/dot-string-template-outlet.directive.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    ViewContainerRef
+} from '@angular/core';
+
+class DotStringTemplateOutletContext {
+    $implicit: unknown = null;
+}
+
+/**
+ * Structural directive to give the ability of using an input of a component
+ * to accept a string or a TemplateRef<any>, only add a ng-container with the structural directive (`*dotStringTemplateOutlet`)
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/shared/directives/dot-string-template-outlet.directive.ts:18]: Checked.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/ui/dot-experiments-details-table/dot-experiments-details-table.component.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+} from '@angular/core';
+
+import { DotStringTemplateOutletDirective } from '@portlets/shared/directives/dot-string-template-outlet.directive';
+
+/**
+ * Component to display a table with a title and a list of data with its headers
+ * using the index of the object as headers.
+ * Can send the `headers` and `rows` as templates with content projection to
+ * replace the defaults templates.
+ */
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/ui/dot-experiments-details-table/dot-experiments-details-table.component.ts:17]: done.

**Review by hmoreras — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Pull request merged**